backend/include/Shared.hpp

```diff
@@ -126,7 +131,13 @@ struct UserSettings {
                 PLOG_ERROR << "Error loading settings.ini: " << err;
             }
         }
-
+        UserSettings::configVersion = static_cast<int>(ini.GetLongValue("General", "Version", 0));
+        if (UserSettings::configVersion != CONFIG_VERSION) {
+            UserSettings::configVersion = CONFIG_VERSION;
+            PLOG_WARNING << "Settings.ini version mismatch, recreating it";
+            save();
```

Comment:
When this function gets called, you should use the callElectron helper and pass an error message that the config needs to be reset

Reply:
There should be a way to auto-display the settings dialog too, I added that a while back but forget exactly what triggers it.

Reply:
Here's where I did it for the electron settings. Looks like if you send show-settings to the main window once the failure bubbles up that should do the trick.

https://vscode.dev/github/neilenns/TrackAudio/blob/neilenns/issue212/src/main/index.ts#L377-L384

Reply:
![4q6l5G](https://github.com/user-attachments/assets/c0049050-10b5-4869-b1f4-bc10c6e2289b)

@pierr3 @neilenns 

This is both of those implemented.

However, I've had to make a queue inside the index.ts in order to queue events from the TrackAudio C++ callback so that they get sent once the React side has been fully loaded, otherwise these NAPI events get lost as the UserSettings stuff all takes place in the Bootstrap function seems to work fine, though I've put it in the settings-ready ipcHandle as this gets called on Bootstrap React load I mean it's fine though could do with probably being more of  an app-ready.

Reply:
I think this is a sufficient implementation, but good catch on the queue issue